### ISSUE TITLE
[Feature request] Add Fork git instance to PATH on Windows

### ISSUE THREAD

**Issue by hudcap** [ForkIssues/TrackerWin#601]
I use Fork as my main git client. In VSCode, I change the `git.path` setting to the git instance that ships with Fork. The only problem is that the path changes every time git gets updated, since the version number is part of the path, and I have to manually update it in VSCode.
Would you consider an option to add the git location to PATH so that VSCode (and perhaps other IDEs) would find it automatically? Or alternatively, to use a non-changing path to the git instance?

Thank you!

**Comment by DanPristupov** [comment]
You can use `%localappdata%\fork\fork.exe` as a constant path.

**Comment by hudcap** [comment]
Please correct me if I'm wrong, but that looks like the path to the Fork program, whereas I'm looking for a constant path to the git executable shipped with Fork.

**Comment by DanPristupov** [comment]
Sorry, I misread the question. Yes, you are right.

> the path changes every time git gets updated, since the version number is part of the path,

Installing git to a different folder is very important, because it allows us to be sure that the destination folder is available for writing and not locked by some active process.

> Would you consider an option to add the git location to PATH 

I tried to use PATH, but it's usage in Windows is not really reliable. Changing the value will not be visible immediately. It will require restart of the dependent applications (or even Windows).

I think the internal Fork git instance is not intended to be used outside. A better idea could be to use %programfiles%\git\bin\git.exe everywhere (including Fork). It should be also available in PATH.

**Comment by hudcap** [comment]
Just to make sure I understand, you're suggesting installing git on Windows normally and then directing Fork to use the installed git?

**Comment by ImBoop** [comment]
> 
> 
> Sorry, I misread the question. Yes, you are right.
> 
> > the path changes every time git gets updated, since the version number is part of the path,
> 
> Installing git to a different folder is very important, because it allows us to be sure that the destination folder is available for writing and not locked by some active process.
> 
> > Would you consider an option to add the git location to PATH
> 
> I tried to use PATH, but it's usage in Windows is not really reliable. Changing the value will not be visible immediately. It will require restart of the dependent applications (or even Windows).
> 
> I think the internal Fork git instance is not intended to be used outside. A better idea could be to use %programfiles%\git\bin\git.exe everywhere (including Fork). It should be also available in PATH.

It only requires a restart of the parent application to refresh its env vars (e.g., usually explorer). 

You can create a symlink to the git folder and update the symlink every time you guys update (e.g. `%localappdata%\fork\gitInstance\latest\bin` would be added to path, and `latest` would always be a symlink to the latest git folder in Fork.)

 This would allow you to update the git instance without breaking anything that relies on it.

**Comment by Kyrw** [comment]
> You can create a symlink to the git folder and update the symlink every time you guys update (e.g. `%localappdata%\fork\gitInstance\latest\bin` would be added to path, and `latest` would always be a symlink to the latest git folder in Fork.)
> 
> This would allow you to update the git instance without breaking anything that relies on it.

I went here just to make this feature request. Please add a symlink and update it on every release. Also, It would be awesome if on installation we had an option is should add this symlink folder to PATH.

**Comment by ImBoop** [comment]
Any chance of an update on this? Would deeply appreciate it! Thanks!

**Comment by danieldownes** [comment]
Agree, I'm using Fork exclusively for git operations. However I would like other software to find Forks version of git, rather than installing multiple version (which set the PATH environment variable). Or set the PATH manually each time Fork updates its copy of git.

To tweak this feature request, perhaps a button to "set git PATH environment variable to the selected" would be a handy feature. Of course I'm happy to restart as I would need to anyway.

Perfect spot would be here:
![image](https://user-images.githubusercontent.com/1720388/145704479-5926ed3a-455d-4625-baeb-e9b2fa0d5fe8.png)

**Comment by Trezorro** [comment]
I have been having trouble with my SSH keys only working via fork and not in vscode. I'm hoping something like this would be a solution.

**Comment by circleupx** [comment]
This would be a nice feature to have. I dislike having two instances of git. Please consider adding this feature.

**Comment by lukaslabaj** [comment]
+1

**Comment by peter-power-594** [comment]
+1

**Comment by nathan-coleman** [comment]
Hi I'm here just to say that I would also love this!

**Comment by h4okr** [comment]
+1 this would be great

**Comment by AlpyneDreams** [comment]
+1

**Comment by MiragonMx** [comment]
+1
